test/algorithm/test_smooth_quant.py

```diff
@@ -701,5 +705,51 @@ def __iter__(self):
         output2 = q_model.model(input_ids)
 
 
+class TestSqSkipOp(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        class RandDataloader:
+            def __init__(self):
+                pass
+            def __iter__(self):
+                yield torch.rand((1, 4))
+
+        self.linear_dl = RandDataloader()
+
+    @classmethod 
+    def test_sq_skip_op_auto(self):
+        class Model(torch.nn.Module):
+            def __init__(self):
+                super(Model, self).__init__()
+                self.linear0 = nn.Linear(4, 4, bias=False)
+                self.layernorm1 = nn.LayerNorm(4)
+                self.linear1 = nn.Linear(4, 4, bias=False)
+                self.ac1 = nn.ReLU()
+                self.ac2 = nn.LeakyReLU()
+                self.linear2 = nn.Linear(4, 4, bias=True)
+                self.linear3 = nn.Linear(4, 2, bias=True)
+                self.ac3 = nn.Sigmoid()
+
+            def forward(self, x):
+                x = self.linear0(x)
+                x1 = self.layernorm1(x)
+                x_l1 = self.linear1(x1)
+                x_ac1 = self.ac1(x1)
+                x_ac2 = self.ac2(x_ac1)
+                x_l2 = self.linear2(x1)
+                x = x_l1 * x_l2 + x_ac2
```

Comment:
is it possible to simulate add with past kv,  will skip add checking will remove the pattern wrongly

Reply:
also please kindly  double check opt350m and BridgeTower

Reply:
Have validated on opt125m, opt350m, opt1b3, opt2b7, opt6b7, bloom1b7, bloom7b1, gptj, llama, dollyv2-7b, falcon-instruct, mpt-7b-chat.